src/blob/index/simple.rs

```diff
@@ -278,6 +278,28 @@ impl Simple {
             0
         }
     }
+
+    fn mark_all_as_deleted_in_memory(&mut self, key: &[u8]) -> Result<Option<u64>> {
```

Comment:
IMO it's better to omit "_in_memory" in the name of the function, as it's implementation detail, as this function deals with both cases. And it's better to extract inner logic of headers cleanup into separate function, so match branches are smaller and it's easier to read

Reply:
Done